### PR TITLE
Move prometheus rules into chart

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/giantswarm/issues/14241

Moved the alerts using
```sh
$ transform-rules \
    --location g8s-prometheus/helm/g8s-prometheus/prometheus-rules/ \
    --destination prometheus-meta-operatorhelm/prometheus-meta-operator/templates/prometheus-rules \
    --namespace '{{ include "resource.default.namespace" . }}' \
    --filter=team=atlas
```
and manually set `cluster_type: "control_plane"` label on control plane specific alerts.

### CODE REVIEW

File: helm/prometheus-meta-operator/templates/prometheus-rules/prometheus.yml [hunk 1]
```diff
@@ -1,22 +1,19 @@
...
+    {{- include "labels.common" . | nindent 4 }}
+    cluster_type: "control_plane"
 spec:
   groups:
   - interval: 0s
     name: prometheus
     rules:
     - alert: PrometheusCantCommunicateWithTenantAPI
       annotations:
-        description: Prometheus can't communicate with tenant cluster {{ $labels.cluster_id }} Kubernetes API.
```
Comment (omegas27) [inline helm/prometheus-meta-operator/templates/prometheus-rules/prometheus.yml:19]: Why did you remove this cluster id here?

Comment (omegas27) [reply helm/prometheus-meta-operator/templates/prometheus-rules/prometheus.yml:19]: Apart from this, this looks great if tested

Comment (TheoBrigitte) [reply helm/prometheus-meta-operator/templates/prometheus-rules/prometheus.yml:19]: Because this is actually automagically added by alertmanager and opsgenie, and shows up in the alert title.

Comment (omegas27) [reply helm/prometheus-meta-operator/templates/prometheus-rules/prometheus.yml:19]: Cool :)

File: helm/prometheus-meta-operator/templates/prometheus-rules/inhibit.rules.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    rules:
+    - alert: InhibitionOutsideWorkingHours
+      annotations:
+        description: Fires outside working hours.
+      expr: (hour() <= 7 or hour() >= 16) or (day_of_week() > 5 or day_of_week() < 1)
+      labels:
+        area: empowerment
+        nodes_down: "true"
+        outside_working_hours: "true"
+        team: atlas
```
Comment (omegas27) [inline helm/prometheus-meta-operator/templates/prometheus-rules/inhibit.rules.yml:21]: We need to remove this

File: helm/prometheus-meta-operator/templates/prometheus-rules/timesync.rules.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+apiVersion: monitoring.coreos.com/v1
```
Comment (omegas27) [inline helm/prometheus-meta-operator/templates/prometheus-rules/timesync.rules.yml:1]: I think this one should be gone

File: helm/prometheus-meta-operator/templates/prometheus-rules/heartbeat.rules.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  labels:
+    {{- include "labels.common" . | nindent 4 }}
+  name: heartbeat.rules
+  namespace: '{{ include "resource.default.namespace" . }}'
+spec:
+  groups:
+  - name: heartbeat
+    rules:
+    - alert: Heartbeat
+      expr: time() - max(container_start_time_seconds{cluster_type="host", namespace="monitoring", container="prometheus"}) > 20 * 60
```
Comment (omegas27) [inline helm/prometheus-meta-operator/templates/prometheus-rules/heartbeat.rules.yml:14]: We should replace host by control_plane

File: helm/prometheus-meta-operator/templates/prometheus-rules/deployment.rules.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        cancel_if_cluster_status_deleting: "true"
+        cancel_if_cluster_status_updating: "true"
+        severity: page
+        team: atlas
+        topic: controlplane
+    - alert: DeploymentNotSatisfiedChinaAtlas
+      annotations:
+        description: Deployment {{ $labels.exported_namespace}}/{{ $labels.deployment }} is not satisfied.
+        opsrecipe: deployment-not-satisfied/
+      expr: kube_deployment_status_replicas_unavailable{cluster_type="host", deployment=~"alertmanager.*|prometheus|prometheus-meta-operator.*|promxy.*", cluster_id=~"axolotl|argali"} > 0
```
Comment (omegas27) [inline helm/prometheus-meta-operator/templates/prometheus-rules/deployment.rules.yml:31]: Same. we should replace host by control_plane

Comment (omegas27) [inline helm/prometheus-meta-operator/templates/prometheus-rules/deployment.rules.yml:31]: Do we need to alert for giraffe?

**Comment by TheoBrigitte** [comment]
Tested on `gaia`

#### control plane
image

#### tenant cluster
image

old prometheus rules in namespace got cleaned up and new ones live in monitoring namespace

```sh
$ k get prometheusrule -A
NAMESPACE    NAME                       AGE
monitoring   alertmanager.rules         8m56s
monitoring   daemonset.rules            8m56s
monitoring   deployment.rules           8m56s
monitoring   fluentbit.rules            8m56s
monitoring   inhibit.rules              8m56s
monitoring   microendpoint.rules        8m56s
monitoring   prometheus-meta-operator   8m56s
monitoring   prometheus.rules           8m56s
```
also still getting heartbeat

image

File: helm/prometheus-meta-operator/templates/prometheus-rules/alertmanager.rules.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  namespace: '{{ include "resource.default.namespace" . }}'
+spec:
+  groups:
+  - name: alertmanager
+    rules:
+    - alert: AlertManagerNotificationsFailing
+      annotations:
+        description: '{{`AlertManager {{ $labels.integration }} notifications are failing.`}}'
+        opsrecipe: alert-manager-notifications-failing/
+      expr: rate(alertmanager_notifications_failed_total{cluster_id!~"argali|axolotl"}[5m]) > 0
```
Comment (omegas27) [inline helm/prometheus-meta-operator/templates/prometheus-rules/alertmanager.rules.yml:18]: Do we need to alert for giraffe as well?

File: helm/prometheus-meta-operator/templates/prometheus-rules/alertmanager.rules.yml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+      labels:
+        area: empowerment
+        severity: page
+        team: atlas
+        topic: monitoring
+    - alert: AlertManagerNotificationsFailingChina
+      annotations:
+        description: '{{`AlertManager {{ $labels.integration }} notifications are failing.`}}'
+        opsrecipe: alert-manager-notifications-failing/
+      expr: rate(alertmanager_notifications_failed_total{cluster_id=~"argali|axolotl"}[5m]) > 0
```
Comment (omegas27) [inline helm/prometheus-meta-operator/templates/prometheus-rules/alertmanager.rules.yml:29]: Do we need to alert for giraffe as well?

File: helm/prometheus-meta-operator/templates/prometheus-rules/daemonset.rules.yml [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    cluster_type: "control_plane"
+  name: daemonset.rules
+  namespace: '{{ include "resource.default.namespace" . }}'
+spec:
+  groups:
+  - name: daemonset
+    rules:
+    - alert: ControlPlaneDaemonSetNotSatisfiedAtlas
+      annotations:
+        description: '{{`Daemonset {{ $labels.exported_namespace}}/{{ $labels.daemonset }} is not satisfied.`}}'
```
Comment (omegas27) [inline helm/prometheus-meta-operator/templates/prometheus-rules/daemonset.rules.yml:16]:
```suggestion
        description: '{{`Daemonset {{ $labels.namespace}}/{{ $labels.daemonset }} is not satisfied.`}}'
```

File: helm/prometheus-meta-operator/templates/prometheus-rules/daemonset.rules.yml [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+      expr: kube_daemonset_status_number_unavailable{cluster_type="host", cluster_id!="axolotl|giraffe|argali", daemonset=~"fluentbit"} > 0
```
Comment (omegas27) [inline helm/prometheus-meta-operator/templates/prometheus-rules/daemonset.rules.yml:17]:
```suggestion
      expr: kube_daemonset_status_number_unavailable{cluster_type="control_plane", cluster_id!="axolotl|giraffe|argali", daemonset=~"fluentbit"} > 0
```

File: helm/prometheus-meta-operator/templates/prometheus-rules/alertmanager.rules.yml (hunk 7 above)
Comment (omegas27) [inline helm/prometheus-meta-operator/templates/prometheus-rules/alertmanager.rules.yml:29]:
```suggestion
      expr: rate(alertmanager_notifications_failed_total{cluster_id=~"argali|axolotl|giraffe"}[5m]) > 0
```

File: helm/prometheus-meta-operator/templates/prometheus-rules/heartbeat.rules.yml (hunk 4 above)
Comment (TheoBrigitte) [reply helm/prometheus-meta-operator/templates/prometheus-rules/heartbeat.rules.yml]: this is gone, it was the wrong heartbeat anyway.

File: helm/prometheus-meta-operator/templates/prometheus-rules/deployment.rules.yml (hunk 5 above)
Comment (TheoBrigitte) [reply helm/prometheus-meta-operator/templates/prometheus-rules/deployment.rules.yml:31]: Maybe, but let's do this in another PR.

**Review by omegas27 — APPROVED** [review]
